example-crates/example_migrate_wallet/src/main.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::{BTreeSet, HashSet};
+use std::io::Write;
+
+use bdk::bitcoin::Network;
+use bdk::database::SqliteDatabase;
+use bdk::wallet::{AddressIndex, Wallet};
+
+use bdk_electrum::{electrum_client, BdkElectrumClient};
+use bdk_wallet::rusqlite;
+
+const EXTERNAL: &str = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/0/*)";
+const INTERNAL: &str = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/1/*)";
+const ELECTRUM_URL: &str = "ssl://electrum.blockstream.info:60002";
+const NETWORK: Network = Network::Testnet;
+
+const BDK_DB_PATH: &str = ".bdk-example.sqlite";
+const BDK_WALLET_DB_PATH: &str = ".bdk-wallet-example.sqlite";
+
+// Steps for migrating wallet parameters from the old `bdk` 0.30 to the new `bdk_wallet` 1.0.
+// These steps can be applied to wallets backed by a SQLite database. For example to read an
+// existing database, change `BDK_DB_PATH` above to point to the location of the old database
+// file. You may also want to hard code the remaining parameters (descriptors, network, etc)
+// to fit your setup. Note: because we're migrating to a new database there must not already
+// exist a persisted wallet at the new path `BDK_WALLET_DB_PATH`.
+
+// Usage: `cargo run --bin example_migrate_wallet`
+
+fn main() -> anyhow::Result<()> {
+    // Open old wallet
+    let db = SqliteDatabase::new(BDK_DB_PATH);
+    let old_wallet = Wallet::new(EXTERNAL, Some(INTERNAL), NETWORK, db)?;
+
+    // Get last revealed addresses for each keychain
+    let addr = old_wallet.get_address(AddressIndex::LastUnused)?;
```

Comment:
Should sync `old_wallet` to make sure we have an accurate `AddressIndex::LastUnused` or could use `AddressIndex::New` here which makes sure we are working with the last exposed (possibly not yet used) index from `old_wallet`.